src/uwtools/set_namelist.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
```

Comment:
> we moved away from the idea that we'd want to accept a YAML-formatted string as input
> and decided to go for just providing key/value pairs as command line arguments

Does this mean we do not use a YAML file as a configuration argument and only use the "command line" (i.e. **SHUM=-99.0** and not **--config_file=$PATH_TO_YAMLFILE**)

> implementation here does not allow for the user to specify the section where the updated key should be applied.

Do we mean we need to identify specific sections with in the **Jinja** template?

Reply:
> Does this mean we do not use a YAML file as a configuration argument

No. This means that we don't want to deal with the user providing a string formatted like YAML as a way to configure the tool. That is a feature adopted by some of the Python tools in SRW, and unnecessary given we can provide YAML, environment variables, and individual config key/value pairs. 

> Do we mean we need to identify specific sections with in the Jinja template using the Jinja formatting language?

This morning I responded to this and failed to hit  the "submit" button on that comment. It said:

> I want to apologize. I certainly got ahead of myself a bit here. To fill in a template, we do not actually need the arbitrarily deep number of sections for templating. That solution will be needed with later development, though, when we want to override settings in dictable objects. No need to incorporate that feature at this time.

I will go ahead and submit that comment now, just so that it's in the same place as the original and easier to follow the conversation.

Reply:
I'm still not clear what "user providing a string formatted like YAML as a way to configure the tool" actually means.

Reply:
Below is an example of how the tool in SRW is called. I'm suggesting that we _do not_ want to have this sort of interface for our templater.py. In the example, the `settings` environment variable is a _string_ that satisfies YAML syntax. That _string_ is passed in with the `-u` command line argument and parsed as a YAML string in `fill_jinja_template.py`.

We do very much want `templater.py` to be able to provide a YAML-formatted _file_ as a means to configure the values required in the template.

```
  settings="
starttime: !datetime ${CDATE}
cres: ${CRES}"

  $USHDIR/fill_jinja_template.py -q -u "${settings}" -t "${DIAG_TABLE_TMPL_FP}" -o "${diag_table_fp}" || \
    print_err_msg_exit "
```

Reply:
An aside...I'm making this distinction here about what we do not want because the AC for UW-175 hints at wanting this feature, but we have changed course since that ticket was written, and no longer require that feature. I apologize for that AC having gone unnoticed in PI planning.

Reply:
The code I have here and the functional tests in this PR does not do any of this string stuff as you pointed out, and we and does do the key vales pairs on the command line (with or without a config file) as you said it should, so I'm at a total loss at what the original comment was addressing.